server/mailet/mailetcontainer-camel/src/test/java/org/apache/james/mailetcontainer/impl/JamesMailetContextTest.java

```diff
@@ -0,0 +1,142 @@
+        testee.setDomainList(domainList);
+        testee.setUsersRepository(usersRepository);
+        mailAddress = new MailAddress(USERMAIL);
+    }
+
+    @Test
+    public void isLocalUserShouldBeFalseOnNullUser() {
+        assertThat(testee.isLocalUser(null)).isFalse();
+    }
+
+    @Test
+    public void isLocalServerShouldBeFalseWhenDomainDoNotExist() {
+        assertThat(testee.isLocalServer(DOMAIN_COM)).isFalse();
+    }
+
+    @Test
+    public void isLocalServerShouldBeTrueWhenDomainExist() throws Exception {
+        domainList.addDomain(DOMAIN_COM);
+
+        assertThat(testee.isLocalServer(DOMAIN_COM)).isTrue();
+    }
+
+    @Test
+    public void isLocalUserShouldBeTrueWhenUsernameExist() throws Exception {
+        domainList.addDomain(DOMAIN_COM);
+        usersRepository.addUser(USERMAIL, PASSWORD);
+
+        assertThat(testee.isLocalUser(USERMAIL)).isTrue();
+    }
+
+    @Test
+    public void isLocalUserShouldReturnTrueWhenUsedWithLocalPartAndUserExistOnDefaultDomain() throws Exception {
+        HierarchicalConfiguration configuration = mock(HierarchicalConfiguration.class);
+        when(configuration.getString(eq("defaultDomain"), any(String.class)))
+            .thenReturn(DOMAIN_COM);
+
+        domainList.configure(configuration);
+        domainList.addDomain(DOMAIN_COM);
+        usersRepository.addUser(USERMAIL, PASSWORD);
+
+        assertThat(testee.isLocalUser(USERNAME)).isTrue();
+    }
+
+    @Test
+    public void isLocalUserShouldReturnFalseWhenUsedWithLocalPartAndUserDoNotExistOnDefaultDomain() throws Exception {
+        HierarchicalConfiguration configuration = mock(HierarchicalConfiguration.class);
+        when(configuration.getString(eq("defaultDomain"), any(String.class)))
+            .thenReturn("any");
+
+        domainList.configure(configuration);
+        domainList.addDomain(DOMAIN_COM);
+        usersRepository.addUser(USERMAIL, PASSWORD);
+
+        assertThat(testee.isLocalUser(USERNAME)).isFalse();
+    }
+
+    @Test
+    public void isLocalUserShouldBeFalseWhenUsernameDoNotExist() throws Exception {
+        assertThat(testee.isLocalUser(USERMAIL)).isFalse();
+    }
+
+    @Test
+    public void isLocalEmailShouldBeFalseWhenUsernameDoNotExist() throws Exception {
+        assertThat(testee.isLocalEmail(mailAddress)).isFalse();
+    }
+
+    @Test
+    public void isLocalEmailShouldBeFalseWhenUsernameDoNotExistButDomainExists() throws Exception {
+        domainList.addDomain(DOMAIN_COM);
+
+        assertThat(testee.isLocalEmail(mailAddress)).isFalse();
+    }
+
+    @Test
+    public void isLocalEmailShouldBeTrueWhenUsernameExists() throws Exception {
+        domainList.addDomain(DOMAIN_COM);
+        usersRepository.addUser(USERMAIL, PASSWORD);
+
+        assertThat(testee.isLocalEmail(mailAddress)).isTrue();
+    }
+
+    @Test
+    public void isLocalEmailShouldBeFalseWhenMailIsNull() throws Exception {
+        assertThat(testee.isLocalEmail(null)).isFalse();
+    }
+}
```

Comment:
How is about the others methods? There are so many methods on JamesMailetContext beside of isLocal*